### PR TITLE
chore: Transform anchor test to typescript

### PR DESCRIPTION
**Pull request by fa93hws**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
Refactor anchor test to typescript

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/anchor/__tests__/Anchor.test.tsx [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    }
+    return container;
+  };
+}
+
+function createDiv() {
+  const root = document.createElement('div');
+  document.body.appendChild(root);
+  return root;
+}
```
Comment (fa93hws) [inline components/anchor/__tests__/Anchor.test.tsx:22]: The tests are running in parallel.
Using a shared container may cause race condition...

File: components/anchor/__tests__/Anchor.test.tsx [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+let idCounter = 0;
+const getHashUrl = () => `API-${idCounter++}`;
```
Comment (xrkffgg) [inline components/anchor/__tests__/Anchor.test.tsx:25]:
```suggestion
const getHashUrl = () => `Anchor-API-${idCounter++}`;
```

Will this be better?

Comment (xrkffgg) [reply components/anchor/__tests__/Anchor.test.tsx:25]: Almost the same

File: components/anchor/__tests__/Anchor.test.tsx [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    expect(scrollToSpy).toHaveBeenLastCalledWith(0, 800);
+
+    dateNowMock.mockRestore();
+  });
+
+  it('Anchor onChange prop', async () => {
+    const onChange = jest.fn();
+    const wrapper = mount<Anchor>(
+      <Anchor onChange={onChange}>
+        <Link href="#API1" title="API1" />
```
Comment (xrkffgg) [inline components/anchor/__tests__/Anchor.test.tsx:332]: 最后这个不用 hash 了吗

Comment (fa93hws) [reply components/anchor/__tests__/Anchor.test.tsx:332]: 要的要的，忘记了😂

**Review by xrkffgg — COMMENTED** [review]
提示有冲突，rebase 下 master ， force push 下

**Comment by xrkffgg** [comment]
今晚要发新版，刚才 master 合了 feature ，你再 rebase 下吧 😂

然后把 changelog 和 selfcheck 都 填一下

**Review by xrkffgg — APPROVED** [review]

**Comment by fa93hws** [comment]
> 今晚要发新版，刚才 master 合了 feature ，你再 rebase 下吧 😂
> 
> 然后把 changelog 和 selfcheck 都 填一下

Test也要写上去吗

**Comment by xrkffgg** [comment]
image

changelog 写一下

下面4个 没问题 就勾选一下

**Comment by xrkffgg** [comment]
如果 对用户 没有太大影响的话。

可以 在changelog 写　ｎｏｎｅ

在 💡 Background and solution 中写一下 你修改的内容

**Comment by xrkffgg** [comment]
`git push -f`

**Comment by xrkffgg** [comment]
现在这个 PR 乱了，你要不把你改的 复制 出来，重新发个 PR 吧

**Pull request merged**